server/src/test/java/org/elasticsearch/monitor/jvm/HotThreadsTests.java

```
@@ -939,6 +942,55 @@ public void testInnerDetectCPUModeTransportThreads() throws Exception {
         assertThat(innerResult, containsString("org.elasticsearch.monitor.testOther.methodFinal(Some_File:1)"));
     }
 
+    public void testRejectConcurrentDetections() throws Exception {
+        ThreadMXBean mockedMXBean = makeMockMXBeanHelper();
+
+        long[] threadIds = new long[] { 1, 2, 3, 4 }; // Adds up to 10, the intervalNanos for calculating time percentages
+        when(mockedMXBean.getAllThreadIds()).thenReturn(threadIds);
+
+        List<ThreadInfo> allInfos = makeThreadInfoMocksHelper(mockedMXBean, TEST_MOCK_TRANSPORT_THREAD_PREFIX, threadIds, 100_000);
+        List<ThreadInfo> cpuOrderedInfos = List.of(allInfos.get(0), allInfos.get(1), allInfos.get(2), allInfos.get(3));
+        when(mockedMXBean.getThreadInfo(ArgumentMatchers.any(), anyInt())).thenReturn(cpuOrderedInfos.toArray(new ThreadInfo[0]));
+
+        final var barrier = new CyclicBarrier(2);
+        final var backgroundThread = new Thread(() -> {
+            try {
+                new HotThreads().detect(new Writer() {
+                    boolean firstWrite = true;
+
+                    @Override
+                    public void write(char[] cbuf, int off, int len) {
+                        if (firstWrite) {
+                            firstWrite = false;
+                            safeAwait(barrier);
+                            safeAwait(barrier);
+                        }
+                    }
+
+                    @Override
+                    public void flush() {}
+
+                    @Override
+                    public void close() {}
+                });
+            } catch (Exception e) {
+                fail(e);
+            }
+        });
+        backgroundThread.start();
+
+        safeAwait(barrier);
+        try (var stringWriter = new StringWriter()) {
+            new HotThreads().detect(stringWriter);
+            assertEquals(ALREADY_RUNNING_MESSAGE, stringWriter.toString());
+        } catch (Exception e) {
+            fail(e);
```

Why not just let the exception bubble up?